### ISSUE TITLE
R testhat tests fail

### ISSUE THREAD

**Issue by mfansler** [catboost/catboost#2375]
Problem: Observe three failures of the `testthat` tests on the Conda Forge CI builds for version. Common denominator appears to be the `adult` testing data not getting labeled correctly (possibly in the `train` subset).
catboost version: v1.2
Operating System: macOS | Windows | Linux
CPU: x86_64
GPU: N/A

## Description
Working around #2374 with a renaming of `inst/libs/libcatboostr.so` to `inst/libs/libcatboostr.so`, I have enabled `testthat` tests to verify that package is in working order. This leads to the follow failure on all platforms we build for (**linux-64**, **osx-64**, **win-64**)

<details>

```r
══ Failed tests ════════════════════════════════════════════════════════════════
── Error ('test_caret_parameter_tuning.R:46:3'): test caret train and parameter tuning on adult pool ──
Error: Stopping
Backtrace:
    ▆
 1. ├─caret::train(...) at test_caret_parameter_tuning.R:46:2
 2. └─caret:::train.default(...)
── Error ('test_pool.R:208:3'): pool: nan ──────────────────────────────────────
Error in `head.catboost.Pool(first_pool, nrow(first_pool))`: /src/catboost/catboost/R-package/src/catboostr.cpp:434: Dataset slicing error: non-numeric features present, slicing datasets with categorical and text features is not supported
Backtrace:
    ▆
 1. ├─testthat::expect_true(...) at test_pool.R:208:2
 2. │ └─testthat::quasi_label(enquo(object), label, arg = "object")
 3. │   └─rlang::eval_bare(expr, quo_get_env(quo))
 4. ├─base::identical(...)
 5. ├─base::matrix(...)
 6. ├─base::unlist(head(first_pool, nrow(first_pool)))
 7. ├─utils::head(first_pool, nrow(first_pool))
 8. └─catboost:::head.catboost.Pool(first_pool, nrow(first_pool))
── Error ('test_pool.R:233:3'): pool: data.frame vs tibble::tbl_df vs pool ─────
Error in `head.catboost.Pool(pool)`: /src/catboost/catboost/R-package/src/catboostr.cpp:434: Dataset slicing error: non-numeric features present, slicing datasets with categorical and text features is not supported
Backtrace:
    ▆
 1. ├─utils::head(pool) at test_pool.R:233:2
 2. └─catboost:::head.catboost.Pool(pool)

[ FAIL 3 | WARN 7 | SKIP 0 | PASS 76 ]
```

</details>

Running [part of the test code](https://github.com/catboost/catboost/blob/9b286dc62729337eb5a5712f104c0b974f437cbb/catboost/R-package/tests/testthat/test_caret_parameter_tuning.R#L46) locally (macOS), I am able to further generate messages reporting that the label "-1" was only found in the `test` data and not in the `train` data. This appears to 

<details>

```r
> report <- caret::train(
    X_train,
    as.factor(make.names(y_train)),
    method = catboost::catboost.caret,
    preProc = NULL,
    tuneGrid = grid,
    trControl = fit_control,
    test_pool = test_pool
  )
Custom logger is already specified. Specify more than one logger at same time is not thread safe.Custom logger is already specified. Specify more than one logger at same time is not thread safe.Custom logger is already specified. Specify more than one logger at same time is not thread safe.Custom logger is already specified. Specify more than one logger at same time is not thread safe.Custom logger is already specified. Specify more than one logger at same time is not thread safe.Something is wrong; all the Accuracy metric values are missing:
    Accuracy       Kappa    
 Min.   : NA   Min.   : NA  
 1st Qu.: NA   1st Qu.: NA  
 Median : NA   Median : NA  
 Mean   :NaN   Mean   :NaN  
 3rd Qu.: NA   3rd Qu.: NA  
 Max.   : NA   Max.   : NA  
 NA's   :1     NA's   :1    
Error: Stopping
In addition: Warning messages:
1: model fit failed for Resample1: depth=4, learning_rate=0.1, iterations=10, l2_leaf_reg=0.001, rsm=0.95, border_count=64 Error in catboost.train(pool, test_pool, param) : 
  /Users/zomb-ml-platform-msk/go-agent-21.2.0/pipelines/BuildMaster/catboost.git/catboost/private/libs/algo/data.cpp:196: Dataset test #0 contains class label "-1" that is not present in the learn dataset
 
2: model fit failed for Resample2: depth=4, learning_rate=0.1, iterations=10, l2_leaf_reg=0.001, rsm=0.95, border_count=64 Error in catboost.train(pool, test_pool, param) : 
  /Users/zomb-ml-platform-msk/go-agent-21.2.0/pipelines/BuildMaster/catboost.git/catboost/private/libs/algo/data.cpp:196: Dataset test #0 contains class label "-1" that is not present in the learn dataset
 
3: model fit failed for Resample3: depth=4, learning_rate=0.1, iterations=10, l2_leaf_reg=0.001, rsm=0.95, border_count=64 Error in catboost.train(pool, test_pool, param) : 
  /Users/zomb-ml-platform-msk/go-agent-21.2.0/pipelines/BuildMaster/catboost.git/catboost/private/libs/algo/data.cpp:196: Dataset test #0 contains class label "-1" that is not present in the learn dataset
 
4: model fit failed for Resample4: depth=4, learning_rate=0.1, iterations=10, l2_leaf_reg=0.001, rsm=0.95, border_count=64 Error in catboost.train(pool, test_pool, param) : 
  /Users/zomb-ml-platform-msk/go-agent-21.2.0/pipelines/BuildMaster/catboost.git/catboost/private/libs/algo/data.cpp:196: Dataset test #0 contains class label "-1" that is not present in the learn dataset
 
5: model fit failed for Resample5: depth=4, learning_rate=0.1, iterations=10, l2_leaf_reg=0.001, rsm=0.95, border_count=64 Error in catboost.train(pool, test_pool, param) : 
  /Users/zomb-ml-platform-msk/go-agent-21.2.0/pipelines/BuildMaster/catboost.git/catboost/private/libs/algo/data.cpp:196: Dataset test #0 contains class label "-1" that is not present in the learn dataset
 
6: In nominalTrainWorkflow(x = x, y = y, wts = weights, info = trainInfo,  :
  There were missing values in resampled performance measures.
```

</details>

Omitting the `test_pool` argument avoids the error. All three errors appear to involve the `adult` testing dataset.

Testthat tests need to be updated. It is unclear if the error is with the format of the `adult` dataset, or if the loading method have had a change in API.
